gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/TraversalStrategies.java

```diff
@@ -247,10 +248,13 @@ public static final class GlobalCache {
             put(SubgraphStrategy.class.getSimpleName(), SubgraphStrategy.class);
 
             // finalization
+            put(MatchAlgorithmStrategy.class.getSimpleName(), MatchAlgorithmStrategy.class);
```

Comment:
the strategies you added here were not originally part of the grammar because this list only consisted of strategies that we had explicitly included there. i guess these were added to allow tests to pass. it might not be wrong but if you add them here then they should be tested in the feature tests (and i dont think they are). if you do that then each GLV will likely need changes to support them. and if you do these, i don't see why we wouldn't do all of them, which seems like an inevitability. that inevitability leads to (1) is there a better way for GLVs to support new strategies so that there isn't a huge implementation cost of adding proxy strategy classes and (2) a wider question of what happens when a strategy is deprecated or removed...it dynamically affects the grammar.....

Reply:
I don't have strong opinion about adding strategies to Grammar.
May be we can add something similar to `call()` step, but for strategies?

Reply:
after some thought, i feel more inclined to say that i think that they should all be registered, but i guess you don't need to do that for that PR. for now, maybe just symmetrically add the tests for the strategies you added to the feature tests so that this isn't missed and then create a JIRA for how GLVs might have a more dynamic way to instantiate strategy classes. i don't think it's by way of a grammar change, like `call()` but perhaps something similar to: 

https://github.com/apache/tinkerpop/blob/master/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/TraversalStrategyProxy.java

where there is just one construct in a GLV to construct any strategy. then we just deprecate any existing concrete classes that construct strategies. this also solves the problem for how users can work with their own strategies they might create and register by way of Gremlin Server and for provider strategies that get registered per graph.